Src/IronPython/Modules/Builtin.cs

```diff
@@ -625,10 +625,16 @@ public static object eval(CodeContext/*!*/ context, [NotNull]FunctionCode code,
         public static string hex(object? o) {
             object res = PythonOps.Index(o);
             if (res is BigInteger b) {
-                if (b < 0) {
-                    return "-0x" + (-b).ToString("x");
+                if (b == 0) {
+                    return "0x0";
+                } else if (b < 0) {
+                    var val = (-b).ToString("x");
+                    if (val[0] == '0') val = val.TrimStart('0');
```

Comment:
Seems like a bug in .NET. Should probably file an issue and add a comment here.

Reply:
It does indeed look like a bug in .NET, both Framework and Core.